parquet/src/arrow/async_reader/mod.rs

```diff
@@ -201,10 +194,10 @@ impl ArrowReaderMetadata {
 }
 
 #[doc(hidden)]
-// A newtype used within `ReaderOptionsBuilder` to distinguish sync readers from async
-//
-// Allows sharing the same builder for both the sync and async versions, whilst also not
-// breaking the pre-existing ParquetRecordBatchStreamBuilder API
+/// Newtype (wrapper) used within [`ArrowReaderBuilder`] to distinguish sync readers from async
+///
+/// Allows sharing the same builder different readers while keeping the same
```

Comment:
```suggestion
/// Allows sharing the same builder for different readers while keeping the same
```
The sentence doesn't read right, perhaps this is what is missing?